bindings/wasm/examples/src/1_advanced/6_domain_linkage.ts

```diff
@@ -0,0 +1,178 @@
+
+    // The DID Configuration resource can be made available on `https://foo.example.com/.well-known/did-configuration.json`.
+    let configurationResourceJson = configurationResource.toJSON();
+    console.log("Configuration Resource:", JSON.stringify(configurationResource.toJSON(), null, 2));
+
+    // Now the DID Document links to the Domains through the service, and the Foo domain links to the DID
+    // through the DID Configuration resource. A bidirectional linkage is established.
+    // Note however that bidirectionality is not a hard requirement. It is valid to have a Domain Linkage
+    // credential point to a DID, without the DID having a service that points back.
+
+    // =====================================================
+    // Verification can start from two different places.
+    // The first case answers the question "What DID is this domain linked to?"
+    // while the second answers "What domain is this DID linked to?".
+    // ====================================================
+
+    // =====================================================
+    // → Case 1: starting from domain
+    // =====================================================
+
+    // Fetch the DID Configuration resource (For example using the Fetch API).
+    // Note that according to the specs, the DID Configuration resource must exist
+    // at the origin's root, Well-Known Resource directory.
+    const _configurationUrl = `${domainFoo}/.well-known/did-configuration.json")`;
+
+    // But since the DID Configuration
+    // resource isn't available online in this example, we will simply use the JSON.
+    let fetchedConfigurationResource = DomainLinkageConfiguration.fromJSON(configurationResource);
+
+    // Retrieve the issuers of the Domain Linkage Credentials which correspond to the possibly linked DIDs.
+    let issuers: Array<string> = fetchedConfigurationResource.issuers();
+    const issuerDocument: IotaDocument = await didClient.resolveDid(IotaDID.parse(issuers[0]));
+
+    // Validate the linkage between the Domain Linkage Credential in the configuration and the provided issuer DID.
+    // Validation is succeeds when no error is thrown.
+    DomainLinkageValidator.validateLinkage({
+        domain: domainFoo,
+        configuration: fetchedConfigurationResource,
+        issuer: issuerDocument,
+        validationOptions: CredentialValidationOptions.default(),
+    });
+
+    // =====================================================
+    // → Case 2: starting from a DID
+    // =====================================================
+
+    const didDocument: IotaDocument = await didClient.resolveDid(did);
+
+    // Get the Linked Domain Services from the DID Document.
+    let linkedDomainServices: LinkedDomainService[] = didDocument
+        .service()
+        .filter(service => LinkedDomainService.isValid(service))
+        .map(service => LinkedDomainService.fromService(service));
```

Comment:
``type.prototype` doesn't look right

Reply:
`service.type` returns a `string[]` of the `type` field in the service, it's not the language-level type concept.

Reply:
`service.type` could be a getter, no? I mean whatever is more idiomatic.

`fromService` throws an exception on an invalid service that has `"LinkedDomains"` in its `type` field and that's totally fine, because that would be an actual unexcepted error. One should be able to easily implement the `isValid` behavior by try catching around `fromService`.

I would prefer the simple type string check over the `isValid` approach because `fromService` already does that exact structure check so it wastes time by doing it effectively twice.

Reply:
the problem with your approach is that if a service has "linkedDomains" in it's `type` but it's an invalid linked domain service, it will **not** be filtered out and your code will try to convert it to a LinkedDomainService wich will throw an exception and exit, while `isValid` returns a boolean and allows the filtering earlier, preventing any issues when trying to construct a `LinkedDomainService`.

Reply:
Yes, and I think that's okay. You can still use that to have the isValid behavior by wrapping it in try catch, no?

```typescript
let linkedDomainServices: LinkedDomainService[] = didDocument
        .service()
        .filter(service => service.type.includes("LinkedDomains"))
        .map(service => {
            try {
                LinkedDomainService.fromService(service)
            } catch (ex) {
                // ignore if invalid
            }
        });
```

That doesn't duplicate the structure checking for all services of the document.

In the end I'll leave the decision to you, whatever you think is more idiomatic in TS is fine with me :slightly_smiling_face:.